### PR TITLE
feat(cdp): Avo hog migrator

### PR DESCRIPTION
**Pull request by MarconLP**
## Changes

- Adds avo hog function migrator

## How did you test this code?

added tests

### CODE REVIEW

File: posthog/cdp/templates/avo/template_avo.py [hunk 1]
```diff
@@ -124,3 +125,51 @@
...
+
+        events_to_include = [event.strip() for event in includeEvents.split(",") if event.strip()]
+        events_to_exclude = [event.strip() for event in excludeEvents.split(",") if event.strip()]
+
+        if events_to_include:
+            hf["filters"]["events"] = [
+                {"id": event, "name": event, "type": "events", "order": 0} for event in events_to_include
+            ]
+        elif events_to_exclude:
+            event_string = ", ".join(f"'{event}'" for event in events_to_exclude)
```
Comment (mariusandra) [inline posthog/cdp/templates/avo/template_avo.py:156]: You can use this function to be really sure all is good:

```suggestion
            event_string = ", ".join(escape_hogql_string(event) for event in events_to_exclude)
```

File: posthog/cdp/templates/avo/test_template_avo.py [hunk 2]
```diff
@@ -138,3 +140,71 @@ def test_property_filters(self):
...
+    def test_exclude_events(self):
+        obj = self.get_plugin_config({"excludeEvents": "sign up,page view"})
+        template = TemplateAvoMigrator.migrate(obj)
+        assert template["inputs"] == snapshot(
+            {
+                "apiKey": {"value": "1234567890"},
+                "environment": {"value": "dev"},
+                "appName": {"value": "PostHog"},
+                "excludeProperties": {"value": ""},
+                "includeProperties": {"value": ""},
```
Comment (mariusandra) [inline posthog/cdp/templates/avo/test_template_avo.py:197]: Would've been good to test these two getting a config as well, but we can live without.

**Review by mariusandra — APPROVED** [review]
Some suggestions inline, but overall looking great 👍

File: posthog/cdp/templates/avo/test_template_avo.py (hunk 2 above)
Comment (MarconLP) [reply posthog/cdp/templates/avo/test_template_avo.py:197]: added

Comment (MarconLP) [reply posthog/cdp/templates/avo/test_template_avo.py:199]: added

**Pull request merged**